### DISCUSSION TITLE
How are folks using this in production?

### DISCUSSION THREAD

**Discussion by SHxKM** [nickjj/docker-django-example#39] [Q&A] [answered]
First of all huge kudos to @nickjj for putting this out here. Big fan of _Running in Production_ and not surprised to see this quality code here.

I was wondering, since all the "serverless" providers I checked (Heroku, fly.io, Railway) don't support `docker-compose.yml` (while most of them _do_ support `Dockerfile`s), how are people deploying this stack to production?

**Comment by nickjj** [comment]
Hi,

You can run it on any server that you set up on any cloud provider or hardware you own. DigitalOcean is a popular provider but you can use any cloud hosting provider you're comfortable with.

The project is also compatible with Heroku. Folks have successfully deployed it. Here's an issue with the exact Heroku specific configs they added https://github.com/nickjj/docker-rails-example/discussions/70#discussioncomment-7252423. It's for the Rails example app but it 100% applies to this app too.

I'm not sure about Fly, personally I wouldn't use that service. I've heard nothing but horror stories, especially around unreliability with Postgres.

**Reply by SHxKM** [reply]
Thanks @nickjj. Guess what I’m asking is given that most of those services support Dockerfiles but not docker compose, how’d that work? Also, in case I wanna run each service (or at least Postgres) in its own VM. Will dig some more in  here. Thanks.

**Reply by nickjj** [reply, marked as answer]
The linked discussion from my comment goes over how Heroku handles their own config file which is comparable to a Docker Compose config. This project out of the box currently supports running Postgres or Redis or Celery on the same server or separate servers. You can control which services start up with the `COMPOSE_PROFILES` environment variable and all connection strings are configurable as well.
